pilot/pkg/networking/telemetry/telemetry.go

```diff
@@ -37,6 +38,7 @@ var (
 // BuildStatPrefix builds a stat prefix based on the stat pattern.
 func BuildStatPrefix(statPattern string, host string, subset string, port *model.Port, targetPort int, attributes *model.ServiceAttributes) string {
 	prefix := strings.ReplaceAll(statPattern, serviceStatPattern, shortHostName(host, attributes))
+	prefix = strings.ReplaceAll(prefix, serviceNameStatPattern, attributes.Name)
```

Comment:
attributes only exist for Kubernetes. Please check how shortHostName is implemented

Reply:
should i put there a hostname when there is no attributes?

Reply:
yes

Reply:
Imo is better to replace it with empty string as it is with subset

Reply:
name is equivalent to host for non k8s services like service entries

Reply:
ok